compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -106,6 +106,7 @@ class Compiler {
     List(new ElimErasedValueType,    // Expand erased value types to their underlying implmementation types
          new PureStats,              // Remove pure stats from blocks
          new VCElideAllocations,     // Peep-hole optimization to eliminate unnecessary value class allocations
+         new EtaReduce,              // Reduce eta expansions of pure paths to the underlying function reference
```

Comment:
Any fundamental reason this is after erasure but the existing BetaReduce phase is before erasure?

Reply:
The comment of betaReduce mentions blackbox macros, which would be eliminated after erasure. That might have something to do with it, but I did not dig deeper.

Reply:
EtaReduce is after erasure since a lot of context function wizardry happens before or at erasure so we want to make sure this is all done before we apply the optimization.